### DISCUSSION TITLE
CDN integration

### DISCUSSION THREAD

**Discussion by jkthorne** [luckyframework/lucky#1562] [Feature requests & Ideas]
Since lucky is based on speed it would be nice to have CDN support built in.
I think it would be good to have an adapter style module with some built in hooks for adding headers and purging data.
Some notable CDNs to put on the list are Cloud CDN, Cloud Front, Fastly, Cloudflare.

**Comment by jwoertink** [comment]
You can point your assets to a CDN now https://luckyframework.org/guides/frontend/asset-handling#asset-host I'm not super familiar with setting this kind of stuff up. Usually I just work on a project where it's already setup, but do you think we would need any other hooks or setup? I would imagine it's very unique to each setup outside of pointing to the URL.... 

Or do you mean something that pushes the assets to that CDN? i.e. during deploy, send assets to S3 for cloudfront to pick up?

**Reply by jkthorne** [reply]
there are asset CDNs and then there are ones where you can purge assets based on updates to the database.  Basically if a row is updated you can purge sets of data.
This would be for more dynamic CDNs

**Reply by jwoertink** [reply]
Ah interesting. So more of the front-end like [Varnish](https://varnish-cache.org/) and having that distributed. I started writing something in Elixir a few years back https://github.com/jwoertink/shellac It would be cool to have something like that written in Crystal that could sit in front of your app. It then caches the responses, and serves up the cache for you to send out to your CDN.

**Reply by jkthorne** [reply]
Fastly is a CDN build on varnish.  Other CDNs also have some products build on ideas like Varnish.
